### PR TITLE
[INFRA] Update sphinx conf for newer version, correct warnings

### PR DESCRIPTION
**Pull request by emdupre**
This should allow the doc build to complete on GHA !

### CODE REVIEW

**Comment by emdupre** [comment]
🎉  Not the prettiest (yet) but it works !

File: doc/conf.py [hunk 1]
```diff
@@ -175,98 +172,39 @@
...
-    'patsy': ('http://patsy.readthedocs.io/en/latest/', None),
-    'pandas': ('http://pandas.pydata.org/pandas-docs/stable/', None),
+    "python": (_python_doc_base, None),
+    "numpy": ("https://numpy.org/doc/stable/", None),
+    "scipy": ("https://scipy.github.io/devdocs/", None),
+    "matplotlib": ("https://matplotlib.org/stable/", None),
+    "sklearn": ("https://scikit-learn.org/stable/", None),
+    "nibabel": ("https://nipy.org/nibabel", None),
+    "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
+    "nistats": ("https://nistats.github.io", None),
```
Comment (bthirion) [inline doc/conf.py:191]: Should be removed

**Review by bthirion — APPROVED** [review]
LGTM otherwise

**Comment by bthirion** [comment]
Sorry, where can I see the generated doc ?

**Comment by emdupre** [comment]
> Sorry, where can I see the generated doc ?

Right now it's [in the job artifacts](https://github.com/Parietal-INRIA/fmralign/actions/runs/6029500541?pr=69) ! I can expose it elsewhere, though.

**Comment by emdupre** [comment]
The build error was transient as jax released and [then yanked](https://pypi.org/project/jax/#history) a new version.

**Comment by emdupre** [comment]
@bthirion based on [these artifacts](https://github.com/Parietal-INRIA/fmralign/actions/runs/6032918863) and the green checks, I'm going to go ahead and merge ! Will have (an)other PR(s) to update the style and add content

If you have any concerns, let me know !

**Pull request merged**